docs/source/deployment/nodes_grouping.md

```diff
@@ -0,0 +1,107 @@
+# Node Grouping in Kedro: Pipelines, Tags, and Namespaces
+
+When deploying a Kedro project, grouping nodes effectively is crucial for maintainability, debugging, and execution control. This document provides an overview of three key grouping methods: Pipelines, Tags, and Namespaces, along with a list of their strengths, limitations, best use cases, and links to how to use them.
+
+# Choosing between pipelines, tags, and namespaces in Kedro
+
+## Nodes grouping using pipelines in Kedro
+
+### What works with Kedro
+- If your project contains different pipelines, you can use them as predefined node groupings for deployment.
+- Pipelines can be executed separately in the deployment environment.
+- With the visualisation in Kedro Viz, you can switch to see different pipeline in a more isolated view
+<br>
+![Switching between different pipelines in Kedro Viz](../meta/images/kedro_viz_switching_pipeline.gif)
+
+### What doesn't work with Kedro
+- If you need to create custom node groupings that are different with your existing pipelines, creating new pipelines for them may not be convenient. Instead, you can use alternative grouping methods such as tags or namespaces.
+- You cannot execute more than one pipeline in a single step because the kedro run --pipeline command allows running one pipeline at a time.
+- You can switch between different pipelines in Kedro Viz, but the visualisation does not support collapsing or expanding pipelines within Kedro Viz.
+
+### Best used when
+- Your project is structured in a way that requires executing pipelines separately in the deployment environment.
+
+### Not to use when
+- You need to run more than one pipeline together.
+- You want to use the expand or collapse functionality in Kedro Viz
+
+### How to use
+- Run using:
+  ```bash
+  kedro run --pipeline=<your_pipeline_name>
+  ```
+- More information: [Run a pipeline by name](https://docs.kedro.org/en/stable/nodes_and_pipelines/run_a_pipeline.html#run-a-pipeline-by-name)
+
+---
+
+## Tags
+
+### What works with Kedro
+- You can tag individual nodes or the entire pipeline, allowing flexible execution of specific sections without modifying the pipeline structure.
+- Kedro-Viz provides a clear visualisation of tagged nodes, making it easier to understand.
+<br>
+![Filters Panel in Kedro Viz](../meta/images/kedro_viz_filters_tags.png)
+
+### What doesn't work with Kedro
+- Nodes with the same tag can exist in different pipelines, making debugging and maintaining the codebase more challenging.
+- Executing more than one tag in a single step isn't always possible, and sometimes it will fail to run if a part of pipeline is not populated before at the time of execution.
```

Comment:
Isn't this part of every method though? For example, if I run `kedro run --pipeline data_science` before `data_processing`, it will fail, would it not?

Reply:
yeah i think it happens on tags too, so maybe not worth mentioning it at all

Reply:
maybe i can include as same for pipeline and namespace, you can only run 1 at the time